packages/mitex-web/src/main.ts

```diff
@@ -0,0 +1,122 @@
+  \\f\\relax{x} = \\int_{-\\infty}^\\infty
+    \\f\\hat\\xi\\,e^{2 \\pi i \\xi x}
+    \\,d\\xi`;
+
+  /// Capture compiler load status
+  let compilerLoaded = van.state(false);
+  let fontLoaded = van.state(false);
+  window.$typst$script.then(async () => {
+    compilerLoaded.val = true;
+    await $typst.svg({ mainContent: "" });
+    fontLoaded.val = true;
+  });
+
+  /// Create DOM elements
+  const input_area = textarea({
+    class: "mitex-input",
+    placeholder: "Type LaTeX math equations here",
+    value: srcDefault,
+    autofocus: true,
+    rows: 10,
+  });
+  const copy_template_button = button("Copy with template");
+  const copy_template_with_imports_button = button(
+    "Copy with template and imports"
+  );
+  const output = textarea({
+    class: "mitex-output",
+    readOnly: true,
+    placeholder: "Output",
+    rows: 10,
+  });
+
+  /// The preview component
+  const Preview = (sourceCode: State<string>) => {
+    const svgData = van.state("");
+    console.log(sourceCode);
+    van.derive(async () => {
+      if (fontLoaded.val) {
+        svgData.val = await $typst.svg({ mainContent: sourceCode.val });
+      } else {
+        svgData.val = "";
+      }
+    });
+
+    const content = van.derive(() => {
+      if (!compilerLoaded.val) {
+        return "Loading compiler from CDN...";
+      } else if (!fontLoaded.val) {
+        return "Loading fonts from CDN...";
+      } else {
+        return svgData.val;
+      }
+    });
+
+    console.log(compilerLoaded);
+    return div(
+      { class: "mitex-preview" },
+      div({
+        innerHTML: content,
+      })
+    );
+  };
+
+  const previewTmpl = (out: string) => `#import "@preview/mitex:0.1.0": *
+#set page(width: auto, height: auto, margin: 1em);
+#set text(size: 24pt);
+#math.equation(eval("$" + \`${out}\`.text + "$", mode: "markup", scope: mitex-scope), block: true)
+`;
+
+  /// The source code state
+  const src = van.state(
+    previewTmpl(convert_math(srcDefault, new Uint8Array()))
+  );
+
+  const error = div({ class: "error" });
+  const update_output = () => {
+    try {
+      let convert_res = convert_math(input_area.value, new Uint8Array());
+      src.val = previewTmpl(convert_res);
+      output.value = convert_res;
+      error.textContent = "";
+    } catch (e) {
+      output.value = "";
+      error.textContent = e as string;
+    }
+  };
+  output.onfocus = () => output.select();
+  input_area.oninput = update_output;
+  copy_template_button.onclick = () => {
+    update_output();
+    const res = `#math.equation(eval("$" + \`${output.value}\`.text + "$", mode: "markup", scope: mitex-scope), block: true)`;
+    navigator.clipboard.writeText(res);
+  };
+  copy_template_with_imports_button.onclick = () => {
+    update_output();
+    const res = `#import "@preview/mitex:0.1.0": *
+
+#math.equation(eval("$" + \`${output.value}\`.text + "$", mode: "markup", scope: mitex-scope), block: true)`;
+    navigator.clipboard.writeText(res);
+  };

```

Comment:
these bindings can be merged into there definitions. For example, 
```ts
  const output = textarea({
    ...
    oninput: update_output,
  });
```

Reply:
update_output uses input_area and previewTmpl, which has not been defined when defining buttons and areas.

Reply:
the on focus handler is moved

Reply:
What about using some state bindings and placing them ahead?